Analysis/tfcrig/analysis.py

```diff
@@ -234,21 +264,25 @@ def extract_features_from_session_data(
         # Get trial type
         if "currentTrialType" in msg:
             trial_type = int(msg.split(msg_delimiter)[1])
-            if trial_type not in [0, 1]:
-                print(f"Invalid trial type in: '{msg}'!!!")
-                builtin_print(f" - File: {file_name}")
-                return pd.DataFrame()
+            if trial_type not in [0, 1, 2, 3, 4]:
+                raise ValueError(f"Invalid trial type in: '{msg}'!!!")
 
         # Get all trial types, check for balance. Note that if an
         # imbalance is intentional, this will print false positive
         # error messages
-        if "trialTypes" in msg:
-            trial_types = msg.split(msg_delimiter)[1]
+        if "trialTypes" in msg or check_next_trial_types_message:
```

Comment:
@schaefer30037 not super important but just want to call out that there is an impact to using just `Serial.print` instead of the print wrapper functions in the rig code, in this case getting the set of trial types in the analysis pipeline becomes more complex

Reply:
ahh i see. even more reason to fix the buffer issue